src/Microsoft.ML.AutoML/Experiment/MetricsAgents/MultiMetricsAgent.cs

```diff
@@ -66,7 +66,7 @@ public bool IsModelPerfect(double score)
             }
         }
 
-        public MulticlassClassificationMetrics EvaluateMetrics(IDataView data, string labelColumn)
+        public MulticlassClassificationMetrics EvaluateMetrics(IDataView data, string labelColumn, string groupIdColumn)
```

Comment:
**Suggestion**: Throw an exception if `groupIdColumn` is anything but null. Also add this in the other `EvaluateMetrics` that aren't the `RankingMetricsAgent`.

For the record, as discussed offline, adding the `groupIdColumn` parameter to `IMetricsAgent.EvaluateMetrics()` is somewhat bad design, but it's the best we can do for now, and it isn't as bad. Adding these exceptions would help in ensuring that future AutoML developers don't make the mistake of using the new parameter incorrectly.

Reply:
I agree that it doesn't make sense for the user to provide a groupId in non-ranking tasks. I think my only concern is that the behavior for providing unused column, i.e UserID, is to simply not use it but proceed. If providing an extra column info doesn't prohibit the experiment from running properly, should we be throwing an exception? If so, I think we may need to modify not just the behavior within Ranking, but also within other tasks.

Reply:
Mmm... yeah, I guess you're right and it's better not to add the exception.

---
In reply to: [452962245](https://github.com/dotnet/machinelearning/pull/5263#discussion_r452962245) [](ancestors = 452962245)